File: packages/password-autofill/android/src/main/java/io/capawesome/capacitorjs/plugins/passwordautofill/classes/options/SavePasswordOptions.java
package io.capawesome.capacitorjs.plugins.passwordautofill.classes.options;

import androidx.annotation.NonNull;
import com.getcapacitor.PluginCall;
import io.capawesome.capacitorjs.plugins.passwordautofill.classes.CustomExceptions;

public class SavePasswordOptions {

    @NonNull
    private final String password;

    @NonNull
    private final String username;

    public SavePasswordOptions(@NonNull PluginCall call) throws Exception {
        this.password = getPasswordFromCall(call);
        this.username = getUsernameFromCall(call);
    }

    @NonNull
    public String getPassword() {
        return password;
    }

    @NonNull
    public String getUsername() {
        return username;
    }

    @NonNull
    private static String getPasswordFromCall(@NonNull PluginCall call) throws Exception {
        String password = call.getString("password");
        if (password == null) {
            throw CustomExceptions.PASSWORD_MISSING;
        }
        return password;
    }

    @NonNull
    private static String getUsernameFromCall(@NonNull PluginCall call) throws Exception {
        String username = call.getString("username");
        if (username == null) {
            throw CustomExceptions.USERNAME_MISSING;
        }
        return username;
    }
}

File: packages/password-autofill/ios/Plugin/Classes/Options/SavePasswordOptions.swift
import Foundation
import Capacitor

@objc public class SavePasswordOptions: NSObject {
    let domain: String
    let password: String
    let username: String

    init(_ call: CAPPluginCall) throws {
        self.domain = try SavePasswordOptions.getDomainFromCall(call)
        self.password = try SavePasswordOptions.getPasswordFromCall(call)
        self.username = try SavePasswordOptions.getUsernameFromCall(call)
    }

    private static func getDomainFromCall(_ call: CAPPluginCall) throws -> String {
        guard let domain = call.getString("domain") else {
            throw CustomError.domainMissing
        }
        return domain
    }

    private static func getPasswordFromCall(_ call: CAPPluginCall) throws -> String {
        guard let password = call.getString("password") else {
            throw CustomError.passwordMissing
        }
        return password
    }

    private static func getUsernameFromCall(_ call: CAPPluginCall) throws -> String {
        guard let username = call.getString("username") else {
            throw CustomError.usernameMissing
        }
        return username
    }
}
